tests/test_extensions/test_snippets.py

```diff
@@ -892,17 +892,13 @@ def test_url_nested_file(self, mock_urlopen):
     def test_url_lines(self, mock_urlopen):
         """Test specifying specific lines in a URL."""
 
-        content = []
-        length = 0
-        with open('tests/test_extensions/_snippets/lines.txt', 'rb') as f:
-            for l in f:
-                length += len(l)
-                content.append(l)
+        content = open('tests/test_extensions/_snippets/lines.txt', 'rb').read()
```

Comment:
While I'm indifferent as to whether `open` or `with open` is used, if `with` is not used, please explicitly close the file handle. `with` will take care of this for you. I understand that Cpython (specifically) will likely garbage collect this due to it not having a reference and going out of scope, but that is not the case with all Python implementations and I would like it to be explicitly closed, whether with `with` or without it.